pkg/slackclient/client_summary.go

```diff
@@ -0,0 +1,53 @@
+package slackclient
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/oncall-scheduler/pkg/opsgenieclient"
+	"github.com/nlopes/slack"
+)
+
+func (c *Client) PostAlertSummary(summary opsgenieclient.AlertSummary) error {
+	attachments := []slack.Attachment{}
+
+	for _, summaryItem := range summary {
+		attachment := c.buildAttachment(summaryItem)
+		attachments = append(attachments, attachment)
+	}
+
+	_, _, err := c.client.PostMessage(
+		c.channel,
+		slack.MsgOptionAsUser(true),
+		slack.MsgOptionText("Today's OpsGenie Alert Summary!", false),
+		slack.MsgOptionAttachments(attachments...),
+	)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (c *Client) buildAttachment(summaryItem opsgenieclient.AlertSummaryItem) slack.Attachment {
```

Comment:
Godoc with a link to the Slack magic would be useful.

Reply:
Bam done